### DISCUSSION TITLE
how to fit dataframe in bertopic?

### DISCUSSION THREAD

**Discussion by dwissaaj** [MaartenGr/BERTopic#717] [Q&A]
I have a question, how to fit() dataframe in bertopic?  I try to change to list but it keep throwing error.

I also create a google colab code in here https://colab.research.google.com/drive/16igxoq4lCKYxDe9ds1QKwRcRFamrIgef#scrollTo=seFIG701ciOG

**Comment by MaartenGr** [comment]
I cannot access your link but what you generally do is convert a certain column in pandas to a list of non-empty strings and the fitting should work. Generally, errors stem from not using a list or having None/NaN values in there.

**Comment by dwissaaj** [comment]
Already solved . But There are a bit confused how can we save the model to html, as a understanding bertopic using plotly to visualize. Can we achieve this?

**Reply by MaartenGr** [reply]
Although BERTopic uses Plotly to visualize some of the output of BERTopic, you cannot save the entire model to HTML. It is generally not possible to model machine learning models in HTML. 

However, you can save the visualizations as HTML. An example:

```python
fig = topic_model.visualize_barchart()
fig.write_html("path/to/file.html")
```

The same works with all other visualization functions in BERTopic since they return Plotly figures.
